packages/flutter/lib/src/material/about.dart

```diff
@@ -472,6 +482,314 @@ class LicensePage extends StatefulWidget {
+                      widget.about,
+                      const Center(child: CircularProgressIndicator()),
+                    ],
+                  );
+              }
+            },
+          ),
+        );
+      },
+    );
+  }
+
+  void _initDefaultDetailPage(_LicenseData data, BuildContext context) {
+    final String packageName = data.packages[widget.selectedId.value ?? 0];
+    final List<int> bindings = data.packageLicenseBindings[packageName];
+    _MasterDetailFlow.of(context).setInitialDetailPage(
+      _DetailArguments(
+        packageName,
+        bindings.map((int i) => data.licenses[i]).toList(growable: false),
+      ),
+    );
+  }
+
+  Widget _packagesList(
+    final BuildContext context,
+    final int selectedId,
+    final _LicenseData data,
+    final bool drawSelection,
+  ) {
+    return ListView(
+      children: <Widget>[
+        widget.about,
+        ...data.packages
+            .asMap()
+            .entries
+            .map<Widget>((MapEntry<int, String> entry) {
+          final String packageName = entry.value;
+          final int index = entry.key;
+          final List<int> bindings = data.packageLicenseBindings[packageName];
+          return _PackageListTile(
+            packageName: packageName,
+            index: index,
+            isSelected: drawSelection && entry.key == (selectedId ?? 0),
+            numberLicenses: bindings.length,
+            onTap: () {
+              widget.selectedId.value = index;
+              _MasterDetailFlow.of(context).openDetailPage(_DetailArguments(
+                packageName,
+                bindings.map((int i) => data.licenses[i]).toList(growable: false),
+              ));
+            },
+          );
+        }),
+      ],
+    );
+  }
+}
+
+class _PackageListTile extends StatelessWidget {
+  const _PackageListTile({
+    Key key,
+    this.packageName,
+    this.index,
+    this.isSelected,
+    this.numberLicenses,
+    this.onTap,
+}) : super(key:key);
+
+  final String packageName;
+  final int index;
+  final bool isSelected;
+  final int numberLicenses;
+  final GestureTapCallback onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    return Ink(
+      color: isSelected ? Theme.of(context).highlightColor : Theme.of(context).cardColor,
+      child: ListTile(
+        title: Text(packageName),
+        subtitle: Text(MaterialLocalizations.of(context).licensesPackageDetailText(numberLicenses)),
+        selected: isSelected,
+        onTap: onTap,
+      ),
+    );
+  }
+}
+
+class _LicenseData {
+  final List<LicenseEntry> licenses = <LicenseEntry>[];
+  final Map<String, List<int>> packageLicenseBindings = <String, List<int>>{};
+  final List<String> packages = <String>[];
+  // Special treatment for the first package since it should be the package
+  // for delivered application.
+  String firstPackage;
+
+  void addLicense(LicenseEntry entry) {
+    for (final String package in entry.packages) {
```

Comment:
Perhaps skip packages with zero licenses?

Reply:
I said below in https://github.com/flutter/flutter/pull/57588#discussion_r443198697 that I would do this. However, going over the code again it really is already the case that each package always has one or more licenses. This is because we only know about a package should it be recorded as a package to which a given license applies.

Obviously the code must not be clear enough as is, so I'll go through this part and added comments to try and clarify it.